src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -938,13 +951,13 @@ int runShellCmd(const std::string& cmd, std::string& output)
         ret = -EINVAL;
     }
 
-    close(stderr_fds[0]);
     dup2(stderr_fds[1], 2);
     std::shared_ptr<FILE> pipe(popen(cmd.c_str(), "r"), pclose);
     close(stderr_fds[1]);
 
     if (pipe == nullptr) {
         std::cout << "ERROR: Failed to run " << cmd << std::endl;
+        close(stderr_fds[0]);
```

Comment:
prone to error.  fds are resources and should be treated as such, use RAII via a smart ptr so close happens automatically at scope exit.